pyramid/tier_fs_test.go

```diff
@@ -172,6 +181,35 @@ func TestInvalidArgs(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestMultipleConcurrentReads(t *testing.T) {
+	// write a single file to lookup later
+	namespace := uuid.New().String()
+	filename := "1/2/file1.txt"
+	content := "hello world!"
+	writeToFile(t, namespace, filename, content)
+
+	// fill the cache so the file is evicted
+	testEviction(t, namespace)
+	readsSoFar := adapter.gets.Get()
+
+	// try to read that file - only a single access to block storage is expected
+	concurrencyLevel := 50
+	adapter.wait = make(chan struct{})
+	var wg sync.WaitGroup
+	for i := 0; i < concurrencyLevel; i++ {
+		wg.Add(1)
+		go func() {
+			checkContent(t, namespace, filename, content)
```

Comment:
I don't understand this test.  https://github.com/treeverse/lakeFS/pull/1008/files#diff-2c0831b4a8b319091f86b0ee82570cb8780ba086dce76558f7e96c1638028ca8R245 says to read this one filename only once, and wait on the others.  And that is be tested  using `Get()`.  So what does the `adapter.wait` do or test?

Reply:
Well, mem adapter is rather fast. If I won't wrap it and use the code here, I'll get inconsistent behaviour.
The first Get to the adapter will read the file and block. Subsequent Gets will have the file on the disk and read it from there.
However if later on I'll introduce a bug that makes the Get call non-blocking, I won't catch it here since the mem adapter is fast enough to return the first Get with the file before the next calls are reading from the disk.